kv/kvserver: TestReplicaTombstone failed

kv/kvserver.TestReplicaTombstone [failed](https://tanzanite.cluster.engflow.com/invocations/default/10a83506-7b7a-4ec9-91b3-46d58825d77e?testReportRun=1&testReportShard=18&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [bcc993d796d03664604bf695e38fd5644d0bc952](https://github.com/cockroachdb/cockroach/commits/bcc993d796d03664604bf695e38fd5644d0bc952):

Fatal error:

```
panic: test timed out after 40m0s
running tests:
	TestReplicaTombstone (39m46s)
	TestReplicaTombstone/lease-type=LeaseExpiration (39m46s)
	TestReplicaTombstone/lease-type=LeaseExpiration/(4)_(4.1)_raft_messages_to_newer_replicaID_ (32m6s)

```
Stack: 

```
goroutine 443901159 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestReplicaTombstone
=== RUN   TestReplicaTombstone/lease-type=LeaseExpiration/(3)_ReplicaGCQueue
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestReplicaTombstone_lease-type=LeaseExpiration_(3)_ReplicaGCQueue3137430958
    test_log_scope.go:76: use -show-logs to present logs inline
    client_replica_test.go:3465: -- test log scope end --
=== RUN   TestReplicaTombstone/lease-type=LeaseExpiration/(3.1)_(5)_replica_GC_queue_and_merge
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestReplicaTombstone_lease-type=LeaseExpiration_(3.1)_(5)_replica_GC_queue_and_merge1456210726
    test_log_scope.go:76: use -show-logs to present logs inline
    client_replica_test.go:3519: -- test log scope end --
=== RUN   TestReplicaTombstone/lease-type=LeaseExpiration/(4)_(4.1)_raft_messages_to_newer_replicaID_
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestReplicaTombstone_lease-type=LeaseExpiration_(4)_(4.1)_raft_messages_to_newer_replicaID_1332823101
    test_log_scope.go:76: use -show-logs to present logs inline
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=18</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicaTombstone.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45043

kv/kvserver.TestReplicaTombstone [failed](https://tanzanite.cluster.engflow.com/invocations/default/5c9b0714-e971-432c-a162-bc020d586966?testReportRun=1&testReportShard=18&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [b1474fa887606008960634b571cf4501efb6281b](https://github.com/cockroachdb/cockroach/commits/b1474fa887606008960634b571cf4501efb6281b):

Fatal error:

```
panic: test timed out after 40m0s
running tests:
	TestReplicaTombstone (39m44s)
	TestReplicaTombstone/lease-type=LeaseExpiration (39m44s)
	TestReplicaTombstone/lease-type=LeaseExpiration/(2)_ReplicaTooOldError (39m42s)

```
Stack: 

```
goroutine 389868043 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestReplicaTombstone
=== RUN   TestReplicaTombstone/lease-type=LeaseExpiration
=== RUN   TestReplicaTombstone/lease-type=LeaseExpiration/(1)_ChangeReplicasTrigger
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestReplicaTombstone_lease-type=LeaseExpiration_(1)_ChangeReplicasTrigger684900110
    test_log_scope.go:76: use -show-logs to present logs inline
    client_replica_test.go:3351: -- test log scope end --
=== RUN   TestReplicaTombstone/lease-type=LeaseExpiration/(2)_ReplicaTooOldError
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestReplicaTombstone_lease-type=LeaseExpiration_(2)_ReplicaTooOldError480818472
    test_log_scope.go:76: use -show-logs to present logs inline
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=18</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicaTombstone.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestReplicaTombstone [failed](https://tanzanite.cluster.engflow.com/invocations/default/0b76d51f-77c2-440e-bdf5-dde4e96d25c4?testReportRun=1&testReportShard=18&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [749b563a7b5957b2db487c6e8e2fdeed8630103d](https://github.com/cockroachdb/cockroach/commits/749b563a7b5957b2db487c6e8e2fdeed8630103d):

Fatal error:

```
panic: test timed out after 40m0s
running tests:
	TestReplicaTombstone (39m42s)
	TestReplicaTombstone/lease-type=LeaseExpiration (39m42s)
	TestReplicaTombstone/lease-type=LeaseExpiration/(4)_(4.1)_raft_messages_to_newer_replicaID_ (37m34s)

```
Stack: 

```
goroutine 289625059 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestReplicaTombstone
=== RUN   TestReplicaTombstone/lease-type=LeaseExpiration/(4)_(4.1)_raft_messages_to_newer_replicaID_
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestReplicaTombstone_lease-type=LeaseExpiration_(4)_(4.1)_raft_messages_to_newer_replicaID_304337849
    test_log_scope.go:76: use -show-logs to present logs inline
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=18</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicaTombstone.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/client_replica_test.go
+++ b/pkg/kv/kvserver/client_replica_test.go
@@ -3309,6 +3309,9 @@ func TestReplicaTombstone(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	testutils.RunValues(t, "lease-type", roachpb.LeaseTypes(),
 		func(t *testing.T, leaseType roachpb.LeaseType) {
+			if leaseType == roachpb.LeaseExpiration {
+				skip.UnderDeadlock(t, "times out")
+			}
 			t.Run("(1) ChangeReplicasTrigger", func(t *testing.T) {
 				defer leaktest.AfterTest(t)()
 				defer log.Scope(t).Close(t)

